pkg/services/ngalert/api/authorization.go

```diff
@@ -2,9 +2,130 @@ package api
 
 import (
 	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/models"
+	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
 	"github.com/grafana/grafana/pkg/web"
 )
 
+var (
+	// Namespaces (aka folder) scopes
+	ScopeNamespace     = "namespaces"
+	ScopeNamespaceAll  = ac.GetResourceAllScope(ScopeNamespace)
+	ScopeNamespaceName = ac.Scope(ScopeNamespace, "title", ac.Parameter(":Namespace"))
+
+	ScopeDatasource     = "datasources"
+	ScopeDatasourcesAll = ac.GetResourceAllScope(ScopeDatasource)
+	ScopeDatasourceID   = ac.Scope(ScopeDatasource, "id", ac.Parameter(":Recipient"))
+
+	// Alerting rules actions
+	ActionAlertingRuleCreate = "alert.rule:create"
+	ActionAlertingRuleRead   = "alert.rule:read"
+	ActionAlertingRuleUpdate = "alert.rule:update"
+	ActionAlertingRuleDelete = "alert.rule:delete"
+
+	// Alerting instances (+silences) actions
+	ActionAlertingInstanceCreate = "alert.instance:create"
+	ActionAlertingInstanceUpdate = "alert.instance:update"
+	ActionAlertingInstanceRead   = "alert.instance:read"
+
+	// Alerting Notification policies actions
+	ActionAlertingNotificationsCreate = "alert.notifications:create"
+	ActionAlertingNotificationsRead   = "alert.notifications:read"
+	ActionAlertingNotificationsUpdate = "alert.notifications:update"
+	ActionAlertingNotificationsDelete = "alert.notifications:delete"
```

Comment:
One minor question. Should we unify to plural or singular names of rules? So either `alert.rules`, `alert.instances`, `alert.notifications` or `alert.rule`, `alert.instance`, `alert.notification`?

Reply:
good point. I will update